wrappers/common/common.go

```diff
@@ -0,0 +1,158 @@
+package common
+
+import (
+	"context"
+	"net/http"
+	"runtime"
+	"strings"
+
+	"github.com/felixge/httpsnoop"
+	"github.com/honeycombio/beeline-go/propagation"
+	"github.com/honeycombio/beeline-go/timer"
+	"github.com/honeycombio/beeline-go/trace"
+	libhoney "github.com/honeycombio/libhoney-go"
+)
+
+type ResponseWriter struct {
+	http.ResponseWriter
+	Status int
+}
+
+func NewResponseWriter(w http.ResponseWriter) *ResponseWriter {
+	return &ResponseWriter{
+		ResponseWriter: httpsnoop.Wrap(w, httpsnoop.Hooks{}),
```

Comment:
It's supposed to help with #18 but ... I think I didn't do it right. Eventually that's what it's there for - to help implement all the interfaces instead of just the ResponseWriter.